lib/streamlit/secrets.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2018-2021 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import threading
+from typing import Any, Optional, Mapping, MutableMapping
+
+import toml
+
+import streamlit as st
+
+_SECRETS_LOCATION = os.path.expanduser(os.path.join(".", ".streamlit", "secrets.toml"))
+
+
+def _maybe_set_environment_variable(k: Any, v: Any) -> None:
+    value_type = type(v)
+    if value_type in (str, int, float):
+        os.environ[k] = str(v)
+
+
+class Secrets(Mapping[str, Any]):
+    """A dict-like class that stores secrets.
+    Parses secrets.toml on-demand. Cannot be externally mutated.
+    """
+
+    def __init__(self):
+        # Our secrets dict. It will be loaded on demand.
+        self._secrets: Optional[MutableMapping[str, Any]] = None
```

Comment:
Does this need to be a `MutableMapping`? I think the stricter `Mapping` type can be used here since `self._secrets` may point to different objects, but the object it points to is never mutated

Reply:
good call - nope